pipelines/build/openjdk10_pipeline.groovy

```diff
@@ -16,20 +18,43 @@ def javaToBuild = "jdk10u"
 def scmVars = null
 Closure configureBuild = null
 def buildConfigurations = null
+Map<String, ?> DEFAULTS_JSON = null
 
 node ("master") {
     scmVars = checkout scm
-    load "${WORKSPACE}/pipelines/build/common/import_lib.groovy"
-    configureBuild = load "${WORKSPACE}/pipelines/build/common/build_base_file.groovy"
-    buildConfigurations = load "${WORKSPACE}/pipelines/jobs/configurations/${javaToBuild}_pipeline_config.groovy"
+
+    // Load defaultsJson. These are passed down from the build_pipeline_generator and is a JSON object containing some default constants.
+    if (!params.defaultsJson || defaultsJson == "") {
+        throw new Exception("[ERROR] No Defaults JSON found! Please ensure the defaultsJson parameter is populated and not altered during parameter declaration.")
+    } else {
+        DEFAULTS_JSON = new JsonSlurper().parseText(defaultsJson) as Map
+    }
+
+    load "${WORKSPACE}/${DEFAULTS_JSON['importLibraryScript']}"
+
+    // Load baseFilePath. This is where build_base_file.groovy is located. It runs the downstream job setup and configuration retrieval services.
+    if (params.baseFilePath) {
+        configureBuild = load "${WORKSPACE}/${baseFilePath}"
+    } else {
+        configureBuild = load "${WORKSPACE}/${DEFAULTS_JSON['baseFileDirectories']['upstream']}"
+    }
+
+    // Load buildConfigFilePath. This is where jdkxx_pipeline_config.groovy is located. It contains the build configurations for each platform, architecture and variant.
+    if (params.buildConfigFilePath) {
+        buildConfigurations = load "${WORKSPACE}/${buildConfigFilePath}"
+    } else {
+        buildConfigurations = load "${WORKSPACE}/${DEFAULTS_JSON['configDirectories']['build']}/${javaToBuild}_pipeline_config.groovy"
+    }
+
 }
 
 // If a parameter below hasn't been declared above, it is declared in the jenkins job itself
-if (scmVars != null && configureBuild != null && buildConfigurations != null) {
+if (scmVars != null || configureBuild != null || buildConfigurations != null) {
```

Comment:
You're saying if *any* of these is not null - is that the intent?  Previous all 3 had to be not null

Reply:
> is that the intent?

Yes. It should have been that to begin with. If any one of these three don't exist, the job will fail later down the line